testutil/testdevice_linux.go

```diff
@@ -0,0 +1,97 @@
+// +build linux
+
+package testutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+type testDevice struct {
+	opts       TestDeviceOpts
+	fileName   string
+	deviceName string
+}
+
+func NewTestDevice(t *testing.T, opts TestDeviceOpts) TestDevice {
```

Comment:
Because this has no utility outside of the btrfs package. The abstraction is arbitrary and useless.

A little copying is better than a little dependency.